runtime/src/bank/partitioned_epoch_rewards/calculation.rs

```diff
@@ -361,7 +360,7 @@ impl Bank {
                             .entry(vote_pubkey)
                             .or_insert(VoteReward {
                                 commission,
-                                vote_account: vote_account.into(),
+                                vote_account: vote_account.account().clone(),
```

Comment:
nit: this change doesn't look related to the PR

Reply:
Inside voteReward, we store AccountSharedData

```
struct VoteReward {
    vote_account: AccountSharedData,
    commission: u8,
    vote_rewards: u64,
}
```

However, when we get the VoteAccount from cache, we are hold &VoteAccount directly with this change (i.e. we no longer clone &VoteAccount to get VoteAccount as before).

```
pub fn get(&self, pubkey: &Pubkey) -> Option<&VoteAccount>
```

Into() works for VoteAccount, because we implement From traits for VoteAccount not for &VoteAccount. 

To make into() work for &VoteAccount, I added From traits for &VoteAccount too. 

I think this is better than ` vote_account.account().clone()`. In future, we can use `into()` directly on &VoteAccount.

Reply:
Ah I didn't realize we only had the reference now. But yeah implementing `From` for `&VoteAccount` sounds good